source/samples/pubsub/PubSubFeature.cpp

```diff
@@ -30,6 +34,54 @@ string PubSubFeature::getName()
     return "Pub Sub Sample";
 }
 
+bool PubSubFeature::createPubSub(const PlainConfig &config, std::string absFilePath)
+{
+    std::string pubFileDir = FileUtils::ExtractParentDirectory(absFilePath);
+    if (!FileUtils::DirectoryExists(pubFileDir))
+    {
+        // Create an empty directory with the expected permissions.
+        FileUtils::CreateDirectoryWithPermissions(pubFileDir.c_str(), S_IRWXU | S_IRGRP | S_IROTH | S_IXOTH);
```

Comment:
This function returns `true/false` on `success/failure`. You have to test for return and propagate back to caller.

```
if (!FileUtils::CreateDirectoryWithPermissions(pubFileDir.c_str(), S_IRWXU | S_IRGRP | S_IROTH | S_IXOTH))
{
    return false;
}
```

Reply:
done